### PR TITLE
[shelly] Support for Shelly Vintage, Bugfixes

### PR DESCRIPTION
**Pull request by markus7017**
This PR adds support for Shelly Vintage and fixes a problem with updating the lastUpdate channel for sensors.

### CODE REVIEW

File: bundles/org.openhab.binding.shelly/README.md [hunk 1]
```diff
@@ -406,8 +407,7 @@ The Dimmer should be calibrated using the Shelly App.
 
 |Group     |Channel      |Type     |read-only|Description                                                            |
 |----------|-------------|---------|---------|-----------------------------------------------------------------------|
-|control   |power        |Switch   |r/w      |Switch light ON/OFF                                                    |
```
Comment (cpmeister) [inline bundles/org.openhab.binding.shelly/README.md:409]: power channel is removed?

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java [hunk 2]
```diff
@@ -521,7 +521,9 @@ public boolean onEvent(String ipAddress, String deviceName, String deviceIndex,
                 if (type.equals(EVENT_TYPE_RELAY)) {
                     group = profile.numRelays <= 1 ? CHANNEL_GROUP_RELAY_CONTROL : CHANNEL_GROUP_RELAY_CONTROL + rindex;
                     int i = Integer.parseInt(deviceIndex);
-                    if ((i >= 0) && (i <= profile.settings.relays.size())) {
+                    logger.debug("{}: deviceIndex={}", thingName, i);
+                    if ((profile.settings.relays != null) && (i >= 0) && (i < profile.settings.relays.size())) {
+                        logger.debug("{}: get relay info[{}", thingName, i);
```
Comment (cpmeister) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:526]:
```suggestion
                    logger.trace("{}: deviceIndex={}", thingName, i);
                    if ((profile.settings.relays != null) && (i >= 0) && (i < profile.settings.relays.size())) {
                        logger.trace("{}: get relay info[{}]", thingName, i);
```

File: bundles/org.openhab.binding.shelly/README.md (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/README.md:409]: yes, as with other lights power and brightness are controlled through the brightness channel. I don‘t like this, but it was requested in a previous review, now it‘s consistent with HUE, Wemo etc.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java (hunk 2 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/handler/ShellyBaseHandler.java:526]: removed

**Comment by markus7017** [comment]
What's ETA for 2.5.5? I have another small change implementing the Watchdog feature (20 lines or so). Should I add it here or create a new PR?

**Comment by cpmeister** [comment]
2.5.5 should be released on the 18th. Please just put the changes in a another PR, I'll review it tomorrow.

**Comment by markus7017** [comment]
> 2.5.5 should be released on the 18th. Please just put the changes in a another PR, I'll review it tomorrow.

ok, could you merge this one today so I could start the next one. I don't want to deal with 3 branches, merging branches is always a challenge for me, still don't have a clear understanding how git handles this resulting in messud-uip PRs like yesterday (sorry for that).

**Pull request merged**